azavea/pfb-network-connectivity
# Treat territories as states; show country in label

## Overview

This resolves #722 and #723, which are small enough and related enough that it seems easier to lump them together.

Changes:
- Add the US territories (Puerto Rico, Guam, American Samoa, Northern Mariana Islands, and US Virgin Islands) to the list of states offered during Neighborhood creation and remove them from the list of countries.
- Add a data migration to convert any existing neighborhoods that are in the territories and entered as non-US into US neighborhoods with the territory as their state.
- In the public site, show country name for non-US cities in the labels where we show state abbreviation for US cities (country abbreviation would be shorter but more confusing, e.g. Canada and California would both say "CA").

### Demo

Before (no labels, territories are countries):
![image](https://user-images.githubusercontent.com/6598836/53665233-d4f26000-3c38-11e9-9550-f5722bb47c80.png)

After:
![image](https://user-images.githubusercontent.com/6598836/53665120-847b0280-3c38-11e9-9ca5-3c2779ecca7d.png)

### Notes

I pointed this PR at the branch for PR #720 to avoid having to reconcile migrations later.  I'll rebase and redirect before merging.

## Testing Instructions

- Add a neighborhood in a US territory and import results for it. There's a geometry file for San Juan [here](https://s3.amazonaws.com/production-pfb-storage-us-east-1/neighborhood_boundaries/root/PRI/san-juan.zip) and a results file [here](https://s3.amazonaws.com/production-pfb-storage-us-east-1/results_2019_city_speed/san-juan-pr.zip).
   - The creation form should have Puerto Rico in the "State" list and not in the "Country" list.
- Migrating forward should work but do nothing (unless you already had a neighborhood in the territories).  Migrating backward should convert the San Juan neighborhood back into a non-US city.
- In the Places list, detail, and compare views on the public site
   - Territories entered as states (i.e. post-migration) should show up with their abbreviations just like any other state.
   - Territories entered as countries (pre-migration) and any other non-US cities should show up with their full country name.

Note: a neighborhood will only show up on the public site if it has a COMPLETE job, but for the purposes of this testing neither the geometry nor the results actually matter, so you can create any city you want and make it show up in the list by importing the same geometry and results file under different names.

## Checklist


Resolves #722
Resolves #723
